### PR TITLE
BUG:#357227:Methods showing up under Fields Header

### PR DESCRIPTION
**Pull request by v-haiboz**
BUG:#357227:Methods showing up under Fields Header

### CODE REVIEW

File: ECMA2Yaml/ECMAHelper/Models/VersionedValue.cs [hunk 1]
```diff
@@ -47,7 +48,7 @@ public VersionedValue(HashSet<string> monikers, T value)
             Value = value;
         }
     }
-
+    [Serializable]
```
Comment (TianqiZhang) [inline ECMA2Yaml/ECMAHelper/Models/VersionedValue.cs]: why is there so many `[Serializable]` attributes?

Comment (v-haiboz) [reply ECMA2Yaml/ECMAHelper/Models/VersionedValue.cs]: it is used with the DeepClone method of Namespace class.

Comment (TianqiZhang) [reply ECMA2Yaml/ECMAHelper/Models/VersionedValue.cs]: I see, how about just make use of the [MemberwiseClone](https://docs.microsoft.com/en-us/dotnet/api/system.object.memberwiseclone?view=net-5.0#System_Object_MemberwiseClone) method on `Type`? For any `List` or `Dictionary` properties, we don't need to deep copy all the values, just cloning the collection itself will be enough. Something like:
```csharp
        public Type DeepCopy()
        {
            Type rval = (Type) MemberwiseClone();
            rval.BaseTypes = new List<BaseType>(BaseTypes);
            rval.InheritanceChains = new List<VersionedCollection<string>>(InheritanceChains);
            rval.InheritedMembers = new Dictionary<string, VersionedString>(InheritedMembers);
            // go on for all collection properties
            return rval;
        }
```

File: ECMA2Yaml/ECMAHelper/Models/ECMAStore.cs [hunk 2]
```diff
@@ -48,14 +49,17 @@ public ECMAStore(IEnumerable<Namespace> nsList, FrameworkIndex frameworks)
             InheritanceChildrenByUid = new Dictionary<string, List<VersionedString>>();
             ImplementationParentsByUid = new Dictionary<string, List<VersionedString>>();
             ImplementationChildrenByUid = new Dictionary<string, List<VersionedString>>();
-            InheritDocItemsByUid = new Dictionary<string, string>();
```
Comment (TianqiZhang) [inline ECMA2Yaml/ECMAHelper/Models/ECMAStore.cs]: This property is used by other methods, please don't delete it.

File: ECMA2Yaml/ECMAHelper/Models/TypeForwarding.cs [hunk 3]
```diff
@@ -8,7 +8,6 @@ public class TypeForwarding
         public AssemblyInfo From { get; set; }
         public AssemblyInfo To { get; set; }
     }
-
```
Comment (TianqiZhang) [inline ECMA2Yaml/ECMAHelper/Models/TypeForwarding.cs]: no need to update this file.

File: ECMA2Yaml/ECMAHelper/Models/ECMAStore.cs [hunk 4]
```diff
@@ -91,6 +95,88 @@ public void Build()
...
+            {
+                var listType = new List<Type>();
+                item.Types.ForEach(type =>
+                {
+                    item.Types.ForEach(m => {
+                        var classMonikers = m.BaseTypes.Where(bt => bt.Name == "System.Object");
+                        enumCovertClassMoniker = classMonikers.SelectMany(o => o.Monikers).ToList();
+                    });
+
+                    if (_frameworks == null && _frameworks.DocIdToFrameworkDict == null)
```
Comment (TianqiZhang) [inline ECMA2Yaml/ECMAHelper/Models/ECMAStore.cs]:
```suggestion
                    if (_frameworks?.DocIdToFrameworkDict == null)
```

File: ECMA2Yaml/ECMAHelper/Models/ECMAStore.cs [hunk 5]
```diff
@@ -91,6 +95,88 @@ public void Build()
...
+                    {
+                        if (_frameworks.DocIdToFrameworkDict[m.DocId].Any(moniker => enumCovertClassMoniker.Contains(moniker)))
+                        {
+                            members.Add(m);
+                        }
+                    }
+
+                    type.Members = members;
+                    members = new List<Member>();
+                    foreach (var m in type.Overloads)
```
Comment (TianqiZhang) [inline ECMA2Yaml/ECMAHelper/Models/ECMAStore.cs]: Overloads is not built yet at this point, so you can ignore them now. An enum type won't need overloads anyway.

**Review by luyajun0205 — APPROVED** [review]

**Review by TianqiZhang — APPROVED** [review]

**Pull request merged**